benchbuild/environments/entrypoints/cli.py

```diff
@@ -0,0 +1,231 @@
+        uow = unit_of_work.BuildahUnitOfWork()
+        messagebus.handle(cmd, uow)
+
+
+def __pull_sources_in_context(prj: project.Project) -> None:
+    for version in prj.variant.values():
+        src = version.owner
+        src.version(local.cwd, str(version))
+
+
+def create_project_images(projects: ProjectIndex) -> None:
+    """
+    Create project images for all selected projects.
+
+    The image will contain all sources the project requires.
+
+    Args:
+        projects: A project index that contains all reqquested (name, project)
+                  Tuples.
+    """
+    for prj in enumerate_projects(projects):
+        version = make_version_tag(*prj.variant.values())
+        image_tag = make_image_name(f'{prj.name}/{prj.group}', version)
+
+        layers = prj.container
+        layers.context(partial(__pull_sources_in_context, prj))
+        layers.add('.', '/app')
+        layers.env(
+            BB_BUILD_DIR='/app',
+            BB_TMP_DIR='/app',
+            BB_PLUGINS_PROJECTS=f'["{prj.__module__}"]'
+        )
+        layers.workingdir('/app')
+
+        cmd = commands.CreateImage(image_tag, layers)
+        uow = unit_of_work.BuildahUnitOfWork()
+        messagebus.handle(cmd, uow)
+
+
+def enumerate_experiments(
+    experiments: ExperimentIndex, projects: ProjectIndex
+) -> tp.Generator[experiment.Experiment, None, None]:
+    for exp_class in experiments.values():
+        prjs = list(enumerate_projects(projects))
+        yield exp_class(projects=prjs)
+
+
+def create_experiment_images(
+    experiments: ExperimentIndex, projects: ProjectIndex
+) -> None:
+    """
+    Create experiment images for all selected experiments.
+
+    This spawns new container images for each project assigned to the
+    experiment. The project image becomes the new base for the experiment
+    image.
+
+    The image will be prepared to run only the given experiment/project
+    combination by default.
+
+    Args:
+        experiments: An experiment index that contains all requested
+                     (name, experiment) Tuples.
+        projects: A project index that contains all reqquested (name, project)
+                  Tuples.
+    """
+    for exp in enumerate_experiments(experiments, projects):
+        for prj in exp.projects:
+            version = make_version_tag(*prj.variant.values())
+            base_tag = make_image_name(f'{prj.name}/{prj.group}', version)
+            image_tag = make_image_name(
+                f'{exp.name}/{prj.name}/{prj.group}', version
+            )
+
+            image = declarative.ContainerImage().from_(base_tag)
+            image.extend(exp.container)
+            image.env(BB_PLUGINS_EXPERIMENTS=f'["{exp.__module__}"]')
+            verbosity = int(settings.CFG['verbosity'])
+            image.env(BB_VERBOSITY=f'{verbosity}')
+
+            image.entrypoint('benchbuild', 'run', '-E', exp.name, str(prj.id))
+
+            cmd = commands.CreateImage(image_tag, image)
```

Comment:
Buildah only allows lower-case repository names and thus fails on upper-case Experiment names.
A quick fix would be to convert the experiment names to lower-case here.
The same goes for project names and project groups.

This also affects `create_project_images`.

Reply:
Thanks, I did not know that :-)

Reply:
I also only know because of the exception I got ;)

Reply:
should be fixed in the buildah/podman adapters instead of the cli